hs/src/Reach/Connector/ALGO.hs

```diff
@@ -34,9 +34,27 @@ import Reach.Util
 import Reach.Warning
 import Safe (atMay)
 import Text.Read
+import Generics.Deriving ( Generic )
+import Control.Monad.Identity
 
 -- import Debug.Trace
 
+-- Errors for ALGO
+
+data AlgoError
+  = Err_TransferNewToken
+  deriving (Eq, ErrorMessageForJson, ErrorSuggestions, Generic)
+
+instance HasErrorCode AlgoError where
+  errPrefix = const "RA"
+  errIndex = \case
+    Err_TransferNewToken {} -> 0
```

Comment:
The testing apparatus should ensure that these errors are covered in the docs too.